### PR TITLE
g-ls 0.25.3 (new formula)

### PR DESCRIPTION
**Pull request by equationzhao**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/g-ls.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Powerful and cross-platform ls"
+  homepage "https://g.equationzhao.space"
+  url "https://github.com/Equationzhao/g/archive/refs/tags/v0.25.3.tar.gz"
+  sha256 "4c4fb3a1790cf4635795925d93f69db624a56f6d9abd8eb9a634c18a7be73f7a"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build"
```
Comment (chenrui333) [inline Formula/g/g-ls.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w")
```

File: Formula/g/g-ls.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    bin.install "g"
+
+    man1.install buildpath.glob("man/*.1.gz")
+  end
+
+  test do
+    output = shell_output("#{bin}/g -v | grep -E 'Version\\s*[0-9.]*' | grep -v 'Go Version' | awk '{print $3}'")
+    assert_match "0.25.3", output
```
Comment (chenrui333) [inline Formula/g/g-ls.rb:19]:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/g -v")
```

File: Formula/g/g-ls.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class GLs < Formula
+  desc "Powerful and cross-platform ls"
+  homepage "https://g.equationzhao.space"
+  url "https://github.com/Equationzhao/g/archive/refs/tags/v0.25.3.tar.gz"
+  sha256 "4c4fb3a1790cf4635795925d93f69db624a56f6d9abd8eb9a634c18a7be73f7a"
+  license "MIT"
```
Comment (chenrui333) [inline Formula/g/g-ls.rb:6]:
```suggestion
  license "MIT"
  head "https://github.com/Equationzhao/g.git", branch: "master"
```

File: Formula/g/g-ls.rb (hunk 1 above)
Comment (Equationzhao) [reply Formula/g/g-ls.rb]: pass the installation test after adding another output args 
```
*std_go_args(output: "g", ldflags: "-s -w")
```

File: Formula/g/g-ls.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  head "https://github.com/Equationzhao/g.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(output: "g", ldflags: "-s -w")
+    bin.install "g"
```
Comment (branchvincent) [inline Formula/g/g-ls.rb:13]: Small nit, otherwise lgtm:
```suggestion
    system "go", "build", *std_go_args(output: bin/"g", ldflags: "-s -w")
```

**Comment by branchvincent** [comment]
Also, if you can squash all your commits into one to satisfy the [commit style](https://docs.brew.sh/Formula-Cookbook#commit), then this should be ready to merge

**Review by branchv — APPROVED** [review]
LGTM thanks @Equationzhao

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@Equationzhao, thanks for your first contribution to homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

File: Formula/g/g-ls.rb (hunk 4 above)
Comment (chenrui333) [reply Formula/g/g-ls.rb]: good call 👍

**Pull request merged**